core/include/vecmem/memory/unique_ptr.hpp

```diff
@@ -17,6 +17,30 @@
 #include "vecmem/vecmem_core_export.hpp"
 
 namespace vecmem {
+namespace details {
+
+/// Helper function for generating a nonexistent, but still "aligned" pointer.
+///
+/// This is literally just here because MSVC is super touchy about type
+/// casting.
+///
+/// @tparam T The type of the pointer
+/// @return The pointer
+///
+template <typename T>
+T get_nonexistent_pointer() {
+    if constexpr (sizeof(T) == 8) {
+        return reinterpret_cast<T>(0xf000ULL);
+    } else if constexpr (sizeof(T) == 4) {
+        return reinterpret_cast<T>(0xf000UL);
+    } else {
+        static_assert(sizeof(T) == 8 || sizeof(T) == 4,
+                      "Unsupported pointer size.");
+    }
```

Comment:
```suggestion
    return reinterpret_cast<T>(0xf000U);
```

Reply:
As crazy as it sounds, the MSVC warning **is** about casting a smaller type into a larger pointer. Yeah... :confused: So using an even shorter literal would be even worse.

I did this very much on purpose here. Not being able to find anything better than this.

Reply:
And after saying all of this, it turns out I am wrong...

I was seeing warnings in the past because I was using a too large literal in the expression. With the literal being `0xf000U`, MSVC is just fine with casting it to whatever length of a pointer.

Okay, this should indeed simplify things a bit. I still will keep the helper function, as it should still make the intention of this code a bit clearer.